src/compiler/checker.ts

```diff
@@ -1107,6 +1107,24 @@ namespace ts {
             return diagnostic;
         }
 
+        function addDeprecatedSuggestion(location: Node, declarations: Node[], arg0: string | number | undefined) {
+            const diagnostic = createDiagnosticForNode(location, Diagnostics._0_is_deprecated, arg0);
+            for (const declaration of declarations) {
+                const deprecatedTag = getJSDocDeprecatedTag(declaration);
+                if (deprecatedTag) {
+                    addRelatedInfo(
+                        diagnostic,
+                        createDiagnosticForNode(deprecatedTag, Diagnostics.The_declaration_was_marked_as_deprecated_here)
+                    );
+                    break;
+                }
+            }
+            //We call `addRelatedInfo()` before adding the diagnostic
+            //to prevent duplicates.

```

Comment:
I don't understand which duplicates you had in mind. Can you elaborate on what you're trying to avoid?

Reply:
I don't understand this either -- when I call addRelativeInfo second, one test suddenly has 9 suggestions instead of 5.